presto-function-namespace-managers/src/test/java/com/facebook/presto/functionNamespace/TestJsonFileBasedFunctionNamespaceManager.java

```diff
@@ -19,50 +19,56 @@
 import com.facebook.presto.functionNamespace.json.JsonFileBasedFunctionNamespaceManagerModule;
 import com.facebook.presto.spi.function.SqlInvokedFunction;
 import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Resources;
 import com.google.inject.Injector;
 import org.testng.annotations.Test;
 
-import java.io.File;
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.StandardCopyOption;
 import java.util.Collection;
 import java.util.Optional;
 
 import static com.facebook.presto.functionNamespace.testing.SqlInvokedFunctionTestUtils.TEST_CATALOG;
-import static org.assertj.core.util.Files.newTemporaryFile;
 import static org.testng.Assert.assertEquals;
 
 public class TestJsonFileBasedFunctionNamespaceManager
 {
     @Test
-    public void testLoadFunctions() throws IOException
+    public void testLoadFunctions()
     {
-        JsonFileBasedFunctionNamespaceManager jsonFileBasedFunctionNameSpaceManager = createFunctionNamespaceManager();
+        // 1. Test loading of a single json file
+        final String jsonFileName = "json_udf_function_definition.json";
+        final int fileFunctionCount = 9;
+
+        JsonFileBasedFunctionNamespaceManager jsonFileBasedFunctionNameSpaceManager = createFunctionNamespaceManager(jsonFileName);
         Collection<SqlInvokedFunction> functionList = jsonFileBasedFunctionNameSpaceManager.listFunctions(Optional.empty(), Optional.empty());
-        assertEquals(functionList.size(), 9);
+        assertEquals(functionList.size(), fileFunctionCount);
+
+        // 2. Test loading of json files in a directory
+        final String jsonDirName = "json_udf_function_definition_dir";
+        final int dirFunctionCount = 7;
```

Comment:
Does this number include overloaded signatures? As an example, are we counting square once or twice?

Does 7 denote the total number of signatures across udf_1.json, udf_2.json and udaf_1.json? I count 5 unique functions and 9 total signatures, what am I missing?

Reply:
This number includes overloaded functions. I've intentionally duplicated some of the signatures to test if they're properly deduped (overwritten).